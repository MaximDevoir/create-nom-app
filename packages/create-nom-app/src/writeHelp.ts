import chalk from 'chalk'

const helpLines = [
  '',
  `Only ${chalk.blue('<project-name>')} is required.`,
  '',
  'Example:',
  `  ${chalk.green('create-nom-app')} ${chalk.blue('my-first-app')}`,
  '',
  `Run ${chalk.green('create-nom-app')} --help to see all options.`,
  '',
  'If you encounter any issues, feel free to report it to:',
  ' > https://github.com/nom-app/create-nom-app/issues'
]

function writeHelp(): void {
  helpLines.forEach(line => {
    console.log(`  ${line}`)
  })
}

export default writeHelp
