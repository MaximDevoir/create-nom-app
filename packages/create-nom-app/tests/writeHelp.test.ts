import writeHelp from '../src/writeHelp'

describe('writeHelp', () => {
  test('display help', () => {
    const spyOnLog = jest.spyOn(console, 'log').mockImplementation()
    const spyOnError = jest.spyOn(console, 'error').mockImplementation()
    writeHelp()
    expect(spyOnLog).toMatchSnapshot()
    expect(spyOnError).toMatchSnapshot()
    spyOnLog.mockClear()
    jest.clearAllMocks()
  })
})
