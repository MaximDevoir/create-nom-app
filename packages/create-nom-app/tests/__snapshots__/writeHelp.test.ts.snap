// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeHelp display help 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "  ",
    ],
    Array [
      "  Only [34m<project-name>[39m is required.",
    ],
    Array [
      "  ",
    ],
    Array [
      "  Example:",
    ],
    Array [
      "    [32mcreate-nom-app[39m [34mmy-first-app[39m",
    ],
    Array [
      "  ",
    ],
    Array [
      "  Run [32mcreate-nom-app[39m --help to see all options.",
    ],
    Array [
      "  ",
    ],
    Array [
      "  If you encounter any issues, feel free to report it to:",
    ],
    Array [
      "   > https://github.com/create-nom-app/create-nom-app/issues",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`writeHelp display help 2`] = `[MockFunction]`;
